stix_shifter_modules/gcp_chronicle/stix_transmission/results_connector.py

```diff
@@ -86,31 +93,56 @@ def create_results_connection(self, search_id, offset, length, metadata=None):
                                     local_result_count += len(response_text['detections'])
                                     result.append(response_text['detections'])
                             else:
-                                return_obj = self.invalid_response(return_obj, response_dict,
-                                                                   next_response[0].status, response_text)
-                                result = []
+                                # recursive calls to handle 429 resource exhaustion exception during pagination
+                                if next_response[0].status == 429:
```

Comment:
why do you need to specifically handle 429?  How would you know server accept the immediate next call? If the server is busy then you should return the error response unless this is a very special API call.

Reply:
We have observed GCP chronicle returns resource exhastion error (status code 429) when we try to run a query with limit more than 10K in CP4S and also when we run multiple queries in parallel. As a result of this error, we were not able to fetch results more than 10K and we received only partial results . On further anaysis , we have observed that there exists an api rate limit of 10 api hit per 60 seconds for results api(list detection api) which causes this resource exhaustion error. Inorder to fetch more results, we have specifically handled this status 429 exception in the connector code. 
In case of other status codes exceptions, corresponding exceptions are thrown in the CP4S.

In command line, we can search with a maximum length of 10K since the resource exhaustion (429 exception) is handled based on metadata value returned in CP4S loop which is not available in command line

Reply:
We have updated the base branch to develop_4.x. kindly let us know your comments.

Reply:
Updated code with proper error responses for 429 exception and recursion calls are removed.